### ISSUE TITLE
zonemaster.net backend crashes intermittently

### ISSUE THREAD

**Issue by sandoche2k** [zonemaster/zonemaster-backend#497]
zonemaster.net crashed on Friday 12/04. It seems it got restarted due to the restart script in the crontab. When tested on 14/04, it was working and on 15/04 morning it was not working. @pierre had to restart explicitly. 

ToDo: 
1. Add monitoring for the backend testing with the GUI
2. Check when the monitoring posts an issue
3. Identify the crash issue
4. If possible, add more debugging messages to provide information when there is a crash

**Comment by matsduf** [comment]
Suggestion: Run the monitoring very frequently and let it log if backend is not working and then restart backend. Do not restart directly in cron. In this way we can see how long time it goes between crashes.

**Comment by mtoma** [comment]
tail -10 zonemaster_backend_testagent.log 

2019-04-16T11:36:34Z [106919] INFO - Test completed: 47590ded860bd5fc
2019-04-16T11:37:27Z [99408] INFO - Test found: 3bb55fc128fcc622
2019-04-16T11:37:27Z [106922] INFO - Test starting: 3bb55fc128fcc622
2019-04-16T11:37:40Z [106922] INFO - Test completed: 3bb55fc128fcc622
2019-04-16T11:38:06Z [99408] INFO - Test found: 945fcb8454ebbeb1
2019-04-16T11:38:06Z [106924] INFO - Test starting: 945fcb8454ebbeb1
2019-04-16T11:38:38Z [99408] INFO - Test found: 301617d86a17b3e2
2019-04-16T11:38:38Z [99408] CRITICAL - Cannot fork: Cannot allocate memory at /usr/local/share/perl5/Parallel/ForkManager.pm line 52.
2019-04-16T11:38:38Z [99408] NOTICE - Daemon terminating
2019-04-16T11:38:53Z [106924] INFO - Test completed: 945fcb8454ebbeb1

**Comment by mtoma** [comment]
The zonemaster instance on zonemaster.net doesn't have the fix for memory leak...

**Comment by mattias-p** [comment]
I like the idea of monitoring. Do you think we could get a log of memory usage over time for the different services?

**Comment by sandoche2k** [comment]
ToDo:
```
Add monitoring for the backend testing with the GUI - In process
Check when the monitoring posts an issue - Related to the previous point
Identify the crash issue - Hope its the memory leak
If possible, add more debugging messages to provide information when there is a crash - Should be part of 2019.2
```

**Comment by sandoche2k** [comment]
I am keeping this issue as it is. while testing on CentOS 7 with MySQL, I did encounter the intermittent crash. But for CentOS with PSQL and Ubuntu with MySQL and PSQL I was not able to reproduce the same.

**Comment by matsduf** [comment]
It is expected to fixed in version v4.0.0.
